MaxText/layers/llama2.py

```diff
@@ -80,6 +90,15 @@ def __call__(
 
     lnx = nn.with_logical_constraint(lnx, ("activation_batch", "activation_length", "activation_embed"))
 
+    if cfg.model_name == "llama2-70b" and cfg.sharding_strategy == "experimental":
+        query_axis_names: AxisNames = (KV_BATCH, LENGTH, KV_HEAD, KV_HEAD_DIM)
```

Comment:
It makes sense key and value have a different AxisNames and sharding than query, even though the different axis are length and head only; however, it's better not to mix them altogether, I'd say keep the query_axis_names as is, and change query's sharding rules to align with what works for 70B, instead of using kv rules on query.

Reply:
Thanks, Morgan.

The main purpose here is to make the query activation sharded as the same as the key and value to decrease the cross chip communication. How about I add a comment here to explain? Otherwise, I need to crease a new axis name for query activation because the sharding of query activation is not the same as the query proj weight and currently they are sharing the same axis name.

Reply:
added comment.

Reply:
I'd prefer we do it properly even it's experimental config, we already have quite a confusion and assumption throughout the codebase and outdated comments. Let's not introduce any new ambiguity and potential issue for later.

Reply:
Agreed with morgan, its fine to add more axis names, which will allow us to be more specific/have more flexibility moving forward in addition to clearer code

Reply:
Could we do the sharding name refactoring in a separate task, as you said the sharding name is already confused and my cl doesn't introduce any degradation? Also if the query activation sharding is the same as the key value sharding, using the same axis name should be fine.

I added a TODO here.

Reply:
I would also like to avoid this model specific axis names - can we instead use the more specific axis names for all models, and its up the config to decide how to shard?

Reply:
I also agree with that we need to have a better sharding name for all models. But given the current timeline of the MLPerf, let's have a short term solution to unblock the optimization first and then apply a more general solution. I also added a TODO here.

Reply:
+1 to move these to config.